illumos-utils/src/zfs.rs

```diff
@@ -229,6 +303,38 @@ impl Zfs {
         Ok(filesystems)
     }
 
+    /// Get information about datasets within a list of zpools / datasets.
+    ///
+    /// This function is similar to [Zfs::list_datasets], but provides a more
+    /// substantial results about the datasets found.
+    ///
+    /// Sorts results and de-duplicates them by name.
+    pub fn get_dataset_properties(
+        datasets: &[String],
+    ) -> Result<Vec<DatasetProperties>, anyhow::Error> {
+        let mut command = std::process::Command::new(ZFS);
+        let cmd = command.args(&["list", "-d", "1", "-rHpo"]);
+
+        // Note: this is tightly coupled with the layout of DatasetProperties
+        cmd.arg("oxide:uuid,name,avail,used,quota,reservation,compression");
```

Comment:
Ohh this is where the format is. What do you think of making the format string a const member on `DatasetProperties` so it's clear?

Reply:
Done!